src/Swarm/TUI/View.hs

```diff
@@ -511,6 +514,25 @@ descriptionTitle e = " " ++ from @Text (e ^. entityName) ++ " "
 descriptionWidget :: AppState -> Entity -> Widget Name
 descriptionWidget s e = padLeftRight 1 (explainEntry s e)
 
+messagesWidget :: GameState -> Widget Name
+messagesWidget gs = viewport MessageViewport Vertical (vBox widgetList)
+ where
+  widgetList = focusNewest . map drawLogEntry' . toList . Seq.sort $ focusedLogs <> messages
+  focusNewest = over (element 1) visible
```

Comment:
Explicitly setting `visible` on the newest log message is what is causing the scrolling not to work.  `visible` specifically demands that the given element is visible, so you cannot scroll it out of the window.  If I remove the `focusNewest` call then scrolling works as expected.

Reply:
Also, isn't `element 1` the *second* newest log message?  That's why you can scroll the window by one line but no more.

Reply:
Hmm, we use it in the other log list, how is it unset then?

I wanted to go back to that lens and try `_1` or head or something, never did I expect it would behave so weirdly. 🤦

Reply:
Fixed :white_check_mark: